app/src/main/java/com/example/africanid/MainActivity.java

```diff
@@ -3,12 +3,64 @@
 import androidx.appcompat.app.AppCompatActivity;
 
 import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
 
 public class MainActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        Button button = findViewById(R.id.buttonS);
+
+        final TextView label1 = findViewById(R.id.textLabel1);
+        final TextView label2 = findViewById(R.id.textLabel2);
+        final TextView txtBirthday = (TextView)findViewById(R.id.textBirthday);
+        final TextView txtGender = (TextView)findViewById(R.id.textGender);
+
+        label1.setVisibility(View.INVISIBLE);
+        label2.setVisibility(View.INVISIBLE);
+        txtBirthday.setVisibility(View.INVISIBLE);
+        txtGender.setVisibility(View.INVISIBLE);
+
+
+        button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                EditText txtBx = (EditText)findViewById(R.id.txtBx);
+
+
+                String idNumber = txtBx.getText().toString();
+
+                if(idNumber.length()==13) {
+                    label1.setVisibility(View.VISIBLE);
+                    label2.setVisibility(View.VISIBLE);
+                    txtBirthday.setVisibility(View.VISIBLE);
+                    txtGender.setVisibility(View.VISIBLE);
+                    String birthDay = idNumber.substring(0, 6);
+                    birthDay = birthDay.substring(4, 6) + "/" + birthDay.substring(2, 4) + "/" + birthDay.substring(0, 2);
+                    txtBirthday.setText(birthDay);
+
+
+                    String gender = idNumber.substring(6, 10);
+                    if (gender.charAt(0) < '5') {
+                        txtGender.setText("Female");
+                    } else
+                        txtGender.setText("Male");
+                }
+                else
+                {
```

Comment:
You are mixing the brackets start and end ({}) in all your code you have :
if (boolean_condition) {
}
and here in the else you have  
else
{
}
for instance.  could please standardize the use of the brackets ?